test/Store.test.js

```diff
@@ -255,6 +298,21 @@ describe('Store', function () {
       }).should.eql(true);
     });
 
+    it('should serialize payloads before sending', function () {
```

Comment:
These tests are *incredible*. Mind adding one more test that serializes/deserializes a date object and checks the input's (date originally stored in action) `getDate` to the output's (date received after deserialization)? Another check in this test that would be good is checking that the two date objects are not triple-equal to each other to confirm that serialization/deserialization indeed happened.